packages/flutter/lib/src/widgets/editable.dart

```diff
@@ -59,97 +55,87 @@ class _KeyboardClientImpl implements mojom.KeyboardClient {
 
   mojom.EditingState get editingState {
     return new mojom.EditingState()
-      ..text = text
-      ..selectionBase = selection.baseOffset
-      ..selectionExtent = selection.extentOffset
-      ..selectionAffinity = mojom.TextAffinity.values[selection.affinity.index]
-      ..selectionIsDirectional = selection.isDirectional
-      ..composingBase = composing.start
-      ..composingExtent = composing.end;
+      ..text = inputValue.text
+      ..selectionBase = inputValue.selection.baseOffset
+      ..selectionExtent = inputValue.selection.extentOffset
+      ..selectionAffinity = mojom.TextAffinity.values[inputValue.selection.affinity.index]
+      ..selectionIsDirectional = inputValue.selection.isDirectional
+      ..composingBase = inputValue.composing.start
+      ..composingExtent = inputValue.composing.end;
   }
 
   void updateEditingState(mojom.EditingState state) {
-    text = state.text;
-    selection = _getTextSelectionFromEditingState(state);
-    composing = new TextRange(start: state.composingBase, end: state.composingExtent);
+    inputValue = new InputValue(
+      text: state.text,
+      selection: _getTextSelectionFromEditingState(state),
+      composing: new TextRange(start: state.composingBase, end: state.composingExtent)
+    );
     onUpdated();
   }
 
+  void clearComposing() {
+    inputValue = new InputValue(
+      text: inputValue.text,
+      selection: inputValue.selection,
+      composing: TextRange.empty
+    );
```

Comment:
Sometimes we have a `copyWith` function on these immutable objects to make code like this future-proof.  This would become: `inputValue = inputValue.copyWith(composing: TextRange.empty)`.  Just slightly more aesthetic.